[BUG] IO issue when the API key is invalid

**Describe the bug**
If no API key is specified or an invalid API key is used, an IO error occurs when a player sends a message.

`[20:09:35 ERROR]: [PixelChat] Failed to classify message from player ichsehedichnicht: Failed to classify message due to an IO issue.
de.pixelmindmc.pixelchat.exceptions.MessageClassificationException: Failed to classify message due to an IO issue.
        at PixelChatGuardian-1.2.0.jar//de.pixelmindmc.pixelchat.utils.APIHelper.classifyMessage(APIHelper.java:93) ~[?:?]
        at PixelChatGuardian-1.2.0.jar//de.pixelmindmc.pixelchat.listener.AsyncPlayerChatListener.checkIfMessageShouldBeBlocked(AsyncPlayerChatListener.java:155) ~[?:?]
        at PixelChatGuardian-1.2.0.jar//de.pixelmindmc.pixelchat.listener.AsyncPlayerChatListener.onAsyncPlayerChat(AsyncPlayerChatListener.java:125) ~[?:?]
        at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:80) ~[paper-api-26.1.2.build.63-stable.jar:?]
        at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:71) ~[paper-api-26.1.2.build.63-stable.jar:?]
        at io.papermc.paper.plugin.manager.PaperEventManager.callEvent(PaperEventManager.java:54) ~[paper-26.1.2.jar:26.1.2-63-711c5de]
        at io.papermc.paper.plugin.manager.PaperPluginManagerImpl.callEvent(PaperPluginManagerImpl.java:131) ~[paper-26.1.2.jar:26.1.2-63-711c5de]
        at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:628) ~[paper-api-26.1.2.build.63-stable.jar:?]
        at io.papermc.paper.adventure.ChatProcessor.post(ChatProcessor.java:378) ~[paper-26.1.2.jar:26.1.2-63-711c5de]
        at io.papermc.paper.adventure.ChatProcessor.process(ChatProcessor.java:85) ~[paper-26.1.2.jar:26.1.2-63-711c5de]
        at net.minecraft.server.network.ServerGamePacketListenerImpl.chat(ServerGamePacketListenerImpl.java:2603) ~[paper-26.1.2.jar:26.1.2-63-711c5de]
        at net.minecraft.server.network.ServerGamePacketListenerImpl.broadcastChatMessage(ServerGamePacketListenerImpl.java:2641) ~[paper-26.1.2.jar:26.1.2-63-711c5de]
        at net.minecraft.server.network.ServerGamePacketListenerImpl.lambda$handleChat$1(ServerGamePacketListenerImpl.java:2385) ~[paper-26.1.2.jar:26.1.2-63-711c5de]
        at net.minecraft.util.FutureChain.lambda$append$1(FutureChain.java:25) ~[paper-26.1.2.jar:26.1.2-63-711c5de]
        at java.base/java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:739) ~[?:?]
        at java.base/java.util.concurrent.CompletableFuture$Completion.run(CompletableFuture.java:503) ~[?:?]
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090) ~[?:?]
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:614) ~[?:?]
        at java.base/java.lang.Thread.run(Thread.java:1474) ~[?:?]
Caused by: java.io.IOException: Server returned HTTP response code: 401 for URL: https://api.groq.com/openai/v1/chat/completions
        at java.base/jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandleAccessor.java:62) ~[?:?]
        at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499) ~[?:?]
        at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:483) ~[?:?]
        at java.base/sun.net.www.protocol.http.HttpURLConnection.getChainedException(HttpURLConnection.java:1751) ~[?:?]
        at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1321) ~[?:?]
        at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1302) ~[?:?]
        at java.base/sun.net.www.protocol.https.HttpsURLConnectionImpl.getInputStream(HttpsURLConnectionImpl.java:223) ~[?:?]
        at PixelChatGuardian-1.2.0.jar//de.pixelmindmc.pixelchat.utils.APIHelper.decodeResponse(APIHelper.java:178) ~[?:?]
        at PixelChatGuardian-1.2.0.jar//de.pixelmindmc.pixelchat.utils.APIHelper.classifyMessage(APIHelper.java:89) ~[?:?]
        ... 18 more
Caused by: java.io.IOException: Server returned HTTP response code: 401 for URL: https://api.groq.com/openai/v1/chat/completions
        at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1700) ~[?:?]
        at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1302) ~[?:?]
        at java.base/java.net.HttpURLConnection.getResponseCode(HttpURLConnection.java:493) ~[?:?]
        at java.base/sun.net.www.protocol.https.HttpsURLConnectionImpl.getResponseCode(HttpsURLConnectionImpl.java:307) ~[?:?]
        at PixelChatGuardian-1.2.0.jar//de.pixelmindmc.pixelchat.utils.APIHelper.classifyMessage(APIHelper.java:79) ~[?:?]
        ... 18 more`

**To Reproduce**
1. Start the server with the plugin fresh installed
2. Write any message

**Expected behavior**
That there are no IO issues and that ChatGuard will not activate if there is no valid API key. 

**Server Version**
- Minecraft Version: 26.1.2
- Plugin Version: Newest build

--- a/src/main/java/de/pixelmindmc/pixelchat/utils/APIHelper.java
+++ b/src/main/java/de/pixelmindmc/pixelchat/utils/APIHelper.java
@@ -27,10 +27,7 @@ import de.pixelmindmc.pixelchat.exceptions.MessageClassificationException;
 import de.pixelmindmc.pixelchat.model.MessageClassification;
 import org.jetbrains.annotations.NotNull;
 
-import java.io.BufferedReader;
-import java.io.IOException;
-import java.io.InputStreamReader;
-import java.io.OutputStream;
+import java.io.*;
 import java.net.HttpURLConnection;
 import java.net.URI;
 import java.net.URISyntaxException;
@@ -77,8 +74,10 @@ public class APIHelper {
                 return processResponse(jsonResponse);
             } else {
                 String errorResponse = decodeResponse(connection);
-                throw new MessageClassificationException("HTTP error code: " + responseCode + ", Error message: " + errorResponse);
+                throw createHttpException(responseCode, errorResponse);
             }
+        } catch (MessageClassificationException e) {
+            throw e; //Re-raise previously thrown exception
         } catch (IOException e) {
             throw new MessageClassificationException("Failed to classify message due to an IO issue.", e);
         } catch (URISyntaxException e) {
@@ -86,6 +85,28 @@ public class APIHelper {
         } catch (Exception e) {
             throw new MessageClassificationException("Failed to classify message due to an unexpected error.", e);
         }
+    }
+
+    /**
+     * Creates a MessageClassificationException based on the HTTP response code and error response
+     *
+     * @param responseCode The response code returned from the API
+     * @param errorResponse The JSON-string error response from the API
+     * @return A MessageClassificationException with the appropriate message
+     */
+    private @NotNull MessageClassificationException createHttpException(int responseCode, @NotNull String errorResponse) {
+        JsonObject jsonObject = JsonParser.parseString(errorResponse).getAsJsonObject();
+        String apiErrorCode = jsonObject.getAsJsonObject("error").get("code").getAsString();
+
+        if (responseCode == HttpURLConnection.HTTP_UNAUTHORIZED && "invalid_api_key".equals(apiErrorCode)) {
+            return new MessageClassificationException("Invalid API key. Please check your configuration.");
+        }
+
+        if (responseCode == HttpURLConnection.HTTP_NOT_FOUND && "model_not_found".equals(apiErrorCode)) {
+            return new MessageClassificationException("Model not found. Please check your configuration.");
+        }
+
+        return new MessageClassificationException("HTTP error code: " + responseCode + ", Error message: " + errorResponse);
     }
 
     /**
@@ -169,7 +190,15 @@ public class APIHelper {
      */
     private @NotNull String decodeResponse(@NotNull HttpURLConnection connection) throws IOException {
         StringBuilder response = new StringBuilder();
-        try (BufferedReader br = new BufferedReader(new InputStreamReader(connection.getInputStream(), StandardCharsets.UTF_8))) {
+        InputStream stream;
+
+        try {
+            stream = connection.getInputStream();
+        } catch (IOException _) {
+            stream = connection.getErrorStream();
+        }
+
+        try (BufferedReader br = new BufferedReader(new InputStreamReader(stream, StandardCharsets.UTF_8))) {
             String responseLine;
             while ((responseLine = br.readLine()) != null) {
                 response.append(responseLine.trim());

